### ISSUE TITLE
[BUG] Cursor moving 2 carets instead of 1

### ISSUE THREAD

**Issue by Sxmourai** [artisticat1/obsidian-latex-suite#204]
# Description
I just installed Latex suite on my vault, and found out that when the plugin is enabled, whenever I press left arrow / right, instead of moving one caret, it moves 2 which is super frustrating

# To Reproduce
Just enable the plugin

# Expected Behavior
Keep 1 caret when moving the cursor / making it configurable

# Additional Info
LatexSuite version: 1.8.3

Windows 11
Obsidian version: 1.4.13

**Comment by Sxmourai** [comment]
I disabled the plugin and re-enabled and now everything seems to be working fine... I will repost If I have the problem back, but I keep the issue opened right ?

**Comment by Sxmourai** [comment]
Okay so I don't really now why but sometimes its back, then it goes away, then it's back again etc... So it's clearly a bug

**Comment by artisticat1** [comment]
Hi! Thanks for raising an issue. Unfortunately, I can't reproduce this; are there any errors in the console? Does it also happen in the sandbox vault?

**Comment by da-the-dev** [comment]
I also encounter this bug regularly. Seems if I enter $$ $$, make changes, then it might break. Restarting Obsidian helps.
Not only arrow key moves are doubled, but also if you hold down Ctrl, you skip 2 words at the same time

**Comment by da-the-dev** [comment]
I got it on camera.
Try typing in this equation:
$$
p(x)=\dfrac{My-N_{x}}{N}
$$

I first typed p(x), then My (with a mistake, but ok), then -N_{x}, then I selected the right hand size and pressed /. Then I typed N and the cursor doubled.

**Comment by da-the-dev** [comment]
I've tested different plugin version, and it seems that 1.8.0 is the first version when the bug occurs. 1.7.4 is good (if I'm not going crazy lol)

**Comment by artisticat1** [comment]
> I got it on camera. Try typing in this equation: $$ p(x)=\dfrac{My-N_{x}}{N} $$
> 
> I first typed p(x), then My (with a mistake, but ok), then -N_{x}, then I selected the right hand size and pressed /. Then I typed N, pressed Enter on accident, then Backspace to remove it, and the cursor doubled.

Unfortunately, I still can't reproduce this. What is the exact sequence of keypresses you are using?

> I've tested different plugin version, and it seems that 1.8.0 is the first version when the bug occurs. 1.7.4 is good (if I'm not going crazy lol)

1.8.0 indeed involved a lot of refactoring, so it's likely that's where the bug first appeared.

**Comment by da-the-dev** [comment]
> What is the exact sequence of keypresses you are using?
Ok, so, key by key (reproduced in sandbox vault)
- New file
- press dm
- p(x)=My-N(press _)x
- select everything from right to left up to but not including =
- press /
- press N
- press Enter
- press Backspace
- Done

If done in this sequence, if you press left arrow key, you'll see the cursor go left in 2 steps intervals

Maybe it is some combo of "_" shortcut with "/" shortcut in $$ $$ mode.

**Comment by da-the-dev** [comment]
Ok, so
1. I've repeated the sequence, looks like my theory of some shortcut bs is correct, because I can delete and add other characters without changing the result. Most likely the issue is with shortcuts 
2. I logged some stuff in console. I repeated the sequence, and whenever I press arrow keys, this is what I get in console

```
app.js:1 RangeError: A selection needs at least one range
    at e.create (app.js:1:323375)
    at TabstopGroup.toEditorSelection (plugin:obsidian-latex-suite:7371:42)
    at isInsideATabstop (plugin:obsidian-latex-suite:7504:77)
    at handleUpdate (plugin:obsidian-latex-suite:8914:12)
    at e.update (app.js:1:487705)
    at e._dispatch (app.js:1:483944)
    at e.dispatch (app.js:1:485722)
    at vg (app.js:1:650771)
    at bg (app.js:1:650859)
    at kg (app.js:1:651024)
```

**Comment by artisticat1** [comment]
Thanks for posting the error log! I still can't reproduce the bug from your instructions (my cursor always moves by the expected 1), but I've tried fixing the error shown in your logs. Can you try this build and let me know if it fixes the issue?

[obsidian-latex-suite-test.zip](https://github.com/artisticat1/obsidian-latex-suite/files/12802584/obsidian-latex-suite-test.zip)

**Comment by da-the-dev** [comment]
> Can you try this build and let me know if it fixes the issue?

I tried in a sandbox vault and I think this version work! No more duplicated movements, and no errors in the console.

**Comment by artisticat1** [comment]
Thanks! :) I've pushed the fix into a new release. This should now be fixed in 1.8.4.

**Comment by Sxmourai** [comment]
Sorry you went to fast I didn't see the discussion.
I'm happy it helped !
